services/ad-tech/src/routes/seller-router.ts

```diff
@@ -0,0 +1,269 @@
+const DSP_HOSTS = [DSP_HOST!, DSP_A_HOST!, DSP_B_HOST!];
+const DSP_ORIGINS = DSP_HOSTS.map((dspHost) => {
+  return new URL(`https://${dspHost}:${EXTERNAL_PORT}`).toString();
+});
+
+// ********************************************************
+// BYOS implementation of Key - Value store
+// ********************************************************
+/** Intializes scoring signals for key value store. */
+const getDefaultScoringSignals = (): string[][] => {
+  const knownAds: string[][] = [];
+  // Ads from Travel advertiser
+  for (const dspHost of DSP_HOSTS) {
+    knownAds.push([
+      new URL(
+        `https://${dspHost}:${EXTERNAL_PORT}/ads?advertiser=${TRAVEL_HOST}&${RENDER_URL_SIZE_MACRO}`,
+      ).toString(),
+      JSON.stringify({
+        label: 'travel',
+      }),
+    ]);
+  }
+  // Ads from Shop advertiser
+  for (const [key, value] of Object.entries(KNOWN_SHOP_ITEM_LABELS_BY_ID)) {
+    for (const dspHost of DSP_HOSTS) {
+      knownAds.push([
+        new URL(
+          `https://${dspHost}:${EXTERNAL_PORT}/ads?advertiser=${SHOP_HOST}&itemId=${key}&${RENDER_URL_SIZE_MACRO}`,
+        ).toString(),
+        JSON.stringify({
+          label: value,
+        }),
+      ]);
+    }
+  }
+  return knownAds;
+};
+
+/** BYOS implementation of seller key value store. */
+const trustedScoringSignalsStore = new KeyValueStore(
+  getDefaultScoringSignals(),
+);
+
+// ********************************************************
+// SSP helper functions
+// ********************************************************
+const getPerBuyerSignals = (
+  buyerSignals: {[key: string]: {[key: string]: string}} = {},
+): {[key: string]: {[key: string]: string}} => {
+  const perBuyerSignals: {[key: string]: {[key: string]: string}} = {};
+  for (const dspOrigin of DSP_ORIGINS) {
+    if (buyerSignals[dspOrigin]) {
+      perBuyerSignals[dspOrigin] = buyerSignals[dspOrigin];
+    } else {
+      perBuyerSignals[dspOrigin] = {
+        'per_buyer_signals': 'per_buyer_signals',
+      };
+    }
+  }
+  return perBuyerSignals;
+};
+
+/** Assembles and returns an auction configuration. */
+const constructAuctionConfig = (context: {
+  useCase?: string;
+  adType?: string;
+  auctionSignals?: {[key: string]: string};
+  buyerSignals?: {[key: string]: {[key: string]: string}};
+}) => {
+  const useCase = context.useCase || 'default';
+  const adType = context.adType || 'display';
+  const {auctionSignals, buyerSignals} = context;
+  /* If `adType` is `video`, set `resolveToConfig` to `false`. This is because
+   * video ads are only supported with iframes. If `resolveToConfig` is set to
+   * `true`, `runAdAuction()` returns a `FencedFrameConfig`, which can only be
+   * rendered in FencedFrames and not iframes.
+   */
+  const resolveToConfig = adType !== 'video';
+  console.log('Constructing auction config', {
+    useCase,
+    adType,
+    auctionSignals,
+    buyerSignals,
+    resolveToConfig,
+  });
+  const auctionConfig = {
```

Comment:
Move this to a template file that allows for more composability. Imagine a future with "auctionConfig.%%USE_CASE%%.template.json"